lib/kernel-manager.coffee

```diff
@@ -98,7 +74,13 @@ class KernelManager
 
 
     getAllKernelSpecs: ->
-        return _.map @_kernelSpecs, 'spec'
+        # TODO(nikita): switch to async and jupyter-js-services API
+        # Note that using sync here is a temporary regression, because it forces
+        # a manual call to update-kernels by the user.
+        # This will be fixed in a later patch, that adds proper async back
+        kernelSpecs = zmq_services.getZMQGateway().getKernelSpecsSync()
+        _.forEach kernelSpecs, (obj, name) -> obj.spec.name = name
```

Comment:
This line gets deleted in the very next patch. It is here only to allow running this patch standalone without applying the next.